apps/orders-service/src/orders-service.service.ts

```diff
@@ -402,6 +402,20 @@ export class OrdersServiceService {
     }
   }
 
+  public async getUserLastOrderStatus (user: string): Promise<any> {
+    const lastOrder = await this.orderRepository
+      .findRaw()
+      .findOne({ user })
+      .sort({ createdAt: -1 })
```

Comment:
Find one wont sort. Get everything and sort it out. Get the first array index. SortedOrder[0]